### ISSUE TITLE
Cannot build with tensorflow-cc

### ISSUE THREAD

**Issue by dtarakanov1** [microsoft/vcpkg#7853]
**Describe the bug**

There is a problem building projects that use the `tensorflow-cc` (version 1.14.0).
`vcpkg` git revision is `94ef325828961dc5c9f721a0eb9fcc5ba8a2b765` (master).
Extra: a small inconvenience during the package installation.

**Environment**

- OS: Windows
- Compiler: MSVC 2017 (toolset v141)

**To Reproduce**

1. echo | vcpkg.exe install tensorflow-cc:x64-windows (NOTE: `echo` here is because after the message `Configuring TensorFlow` the installation freezes and you have to press `<Enter>`)

2. vcpkg.exe integrate install

3. Create a MSVS project with a file containing (turn off precompiled headers):
```C++
#include "tensorflow/core/public/session.h"
int main() {
    tensorflow::GraphDef graphDef;
}
```

4. The build is failed:
```
1>------ Build started: Project: ml-example, Configuration: Debug x64 ------
1>main.cpp
1>d:\ml-example\ml-example\ml-example\main.cpp(1): fatal error C1083: Cannot open include file: 'tensorflow/core/public/session.h': No such file or directory
1>Done building project "ml-example.vcxproj" -- FAILED.
========== Build: 0 succeeded, 1 failed, 0 up-to-date, 0 skipped ==========
```
The include file is not found, because the actual TensorFlow include root directory is `vcpkg\installed\x64-windows\include\tensorflow-external` not `vcpkg\installed\x64-windows\include`.

5. Add `vcpkg\installed\x64-windows\include\tensorflow-external` into the project `Additional Include Directories`.

6. The file build is failed:
```
1>------ Build started: Project: ml-example, Configuration: Debug x64 ------
1>main.cpp
1>d:\vcpkg\installed\x64-windows\include\tensorflow-external\tensorflow\core\public\session.h(22): fatal error C1083: Cannot open include file: 'tensorflow/core/framework/device_attributes.pb.h': No such file or directory
1>Done building project "ml-example.vcxproj" -- FAILED.
========== Build: 0 succeeded, 1 failed, 0 up-to-date, 0 skipped ==========
```

**Expected behavior**

No user input waiting at step 1.
Successful build at step 4 or at least at step 6.

**Failure logs** 

See above.

**Additional context**

\-

**Comment by dan-shaw** [comment]
Thanks for the issue. Tensorflow is still a work in progress, unfortunately.

Currently, TensorFlow does not work on Windows at all, but it one of our top priorities to support this.  It sort of works in Linux, but it uses vendored dependencies #7823 (so you won't be able to use protobuf and other libraries with your tensorflow installation).

For windows, it also uses vendored dependencies, but TensorFlow upstream does not export any of its symbols in Windows so you still won't be able to use the library even if you build it successfully.

**Comment by dtarakanov1** [comment]
Is there an issue I could subscribe in order to get notified about the work progress?

**Comment by Freeverc** [comment]
Failed with such error:

PS D:\ProgramFiles\vcpkg> .\vcpkg.exe install tensorflow-cc:x64-windows
The following packages will be built and installed:
    tensorflow-cc[core]:x64-windows
Starting package 1/1: tensorflow-cc:x64-windows
Building package tensorflow-cc[core]:x64-windows...
CMake Warning at ports/tensorflow-cc/portfile.cmake:3 (message):
  This tensorflow port currently is experimental on Windows and Linux
  platforms.
Call Stack (most recent call first):
  scripts/ports.cmake:94 (include)


-- Using cached D:/ProgramFiles/vcpkg/downloads/tensorflow-tensorflow-v1.14.0.tar.gz
-- Extracting source D:/ProgramFiles/vcpkg/downloads/tensorflow-tensorflow-v1.14.0.tar.gz
-- Applying patch file-exists.patch
-- Applying patch fix-build-error.patch
-- Using source at D:/ProgramFiles/vcpkg/buildtrees/tensorflow-cc/src/v1.14.0-62c7aa4bd8
-- Acquiring MSYS Packages...
-- Acquiring MSYS Packages... OK
-- Configuring TensorFlow

-- Warning: Building TensorFlow can take an hour or more.
CMake Error at scripts/cmake/vcpkg_execute_build_process.cmake:136 (message):
    Command failed: D:/ProgramFiles/vcpkg/downloads/tools/msys2/msys64/usr/bin/bash.exe --noprofile --norc -c "D:/ProgramFiles/vcpkg/downloads/tools/bazel/0.25.2/bazel.exe build --verbose_failures -c opt --python_path=D:/ProgramFiles/vcpkg/downloads/tools/python/python-3.7.3/python.exe --incompatible_disable_deprecated_attr_params=false --define=no_tensorflow_py_deps=true ///tensorflow:libtensorflow_cc.so ///tensorflow:install_headers"
    Working Directory: D:/ProgramFiles/vcpkg/buildtrees/tensorflow-cc/x64-windows-rel
    See logs for more information:
      D:\ProgramFiles\vcpkg\buildtrees\tensorflow-cc\build-x64-windows-rel-err.log

Call Stack (most recent call first):
  ports/tensorflow-cc/portfile.cmake:90 (vcpkg_execute_build_process)
  scripts/ports.cmake:94 (include)


Error: Building package tensorflow-cc:x64-windows failed with: BUILD_FAILED
Please ensure you're using the latest portfiles with `.\vcpkg update`, then
submit an issue at https://github.com/Microsoft/vcpkg/issues including:
  Package: tensorflow-cc:x64-windows
  Vcpkg version: 2019.09.12-nohash

Additionally, attach any relevant sections from the log files above.

**Comment by jgehw** [comment]
@Freeverc your comment is out of topic here because you cannot build the tensorflow-cc vcpkg. This issue however is about not being able to build VC projects using tensorflow-cc once the tensorflow-cc vcpkg has been successfully built. Concerning your issue, please check if it is a duplicate of #12506 which I solved via PR #12546

**Comment by jgehw** [comment]
Fixed in PR #13028. It doesn't solve the vendored dependencies problem, so you still have to add `vcpkg\installed\x64-windows\include\tensorflow-external` and `vcpkg\installed\x64-windows\include\tensorflow-external\src` to your include path, but at least then it works. Note you also have to define `#define NOMINMAX` (because windows.h defines min and max macros which conflict with std::min() and std::max()).

**Comment by PhoebeHui** [comment]
@jgehw, very appreciate your continued assistance in this! I think we can close this now.
